### PR TITLE
Remove __self and __source location from elements

### PR DESCRIPTION
**Pull request by sebmarkbage**
Along with all the places using it like the `_debugSource` on Fiber. This still lets them be passed into `createElement` (and JSX dev runtime) since those can still be used in existing already compiled code and we don't want that to start spreading to DOM attributes.

We used to have a DEV mode that compiles the source location of JSX into the compiled output. This was nice because we could get the actual call site of the JSX (instead of just somewhere in the component). It had a bunch of issues though:

- It only works with JSX.
- The way this source location is compiled is different in all the pipelines along the way. It relies on this transform being first and the source location we want to extract but it doesn't get preserved along source maps and don't have a way to be connected to the source hosted by the source maps. Ideally it should just use the mechanism other source maps use.
- Since it's expensive it only works in DEV so if it's used for component stacks it would vary between dev and prod.
- It only captures the callsite of the JSX and not the stack between the component and that callsite. In the happy case it's in the component but not always.

Instead, we have another zero-cost trick to extract the call site of each component lazily only if it's needed. This ensures that component stacks are the same in DEV and PROD. At the cost of worse line number information.

The better way to get the JSX call site would be to get it from `new Error()` or `console.createTask()` inside the JSX runtime which can capture the whole stack in a consistent way with other source mappings. We might explore that in the future.

This removes source location info from React DevTools and React Native Inspector. The "jump to source code" feature or inspection can be made lazy instead by invoking the lazy component stack frame generation. That way it can be made to work in prod too. The filtering based on file path is a bit trickier.

When redesigned this UI should ideally also account for more than one stack frame.

With this change the DEV only Babel transforms are effectively deprecated since they're not necessary for anything.

### CODE REVIEW

File: packages/shared/ReactElementType.js [hunk 1]
```diff
@@ -23,7 +18,4 @@ export type ReactElement = {
 
   // __DEV__
   _store: {validated: boolean, ...},
-  _self: React$Element<any>,
-  _shadowChildren: any,
```
Comment (sebmarkbage) [inline packages/shared/ReactElementType.js:27]: Not sure what this was about. Goes way back. https://github.com/facebook/react/pull/7504

File: packages/react-devtools-shared/src/__tests__/storeComponentFilters-test.js [hunk 2]
```diff
@@ -242,7 +242,13 @@ describe('Store component filters', () => {
...
     );
 
-    expect(store).toMatchInlineSnapshot(`[root]`);
+    // TODO: Filtering should work on component location.
+    // expect(store).toMatchInlineSnapshot(`[root]`);
+    expect(store).toMatchInlineSnapshot(`
+      [root]
+        ▾ <Component>
+            <div>
+    `);
```
Comment (hoxyq) [inline packages/react-devtools-shared/src/__tests__/storeComponentFilters-test.js:251]: Can you disable the whole test instead and add this TODO at the top?

**Review by hoxyq — COMMENTED** [review]
1. Can we revert the changes for DevTools (except the test) and add $FlowFixMe and TODOs for migration from `_debugSource` field in fiber? So if we end up publishing new patch version of DevTools, the "go to source" feature will still work in DEV for previous versions of React
2. What is the timeline for `console.createTask`? Is this currently in exploration phase?

File: packages/react-devtools-shared/src/__tests__/storeComponentFilters-test.js (hunk 2 above)
Comment (sebmarkbage) [reply packages/react-devtools-shared/src/__tests__/storeComponentFilters-test.js:251]: The issue with that strategy is if it gets fixed then it doesn’t get reenabled.

**Comment by sebmarkbage** [comment]
I don’t think we’d do the console.createTask thing until Chrome fixes the UI and let custom UIs access the data.

For the devtools thing it would be better to do throw hack strategy that we use for prod component stacks.

**Comment by sebmarkbage** [comment]
> Can we revert the changes for DevTools (except the test) and add $FlowFixMe and TODOs for migration from _debugSource field in fiber?

We could but it's a bit annoying to keep that code in sync with other parts of React. I also worry that it'll just not get done and encourage people to stay on old versions instead.

If we did do the lazy thing, it could also work on older versions of React.

**Comment by sebmarkbage** [comment]
@rickhanlonii stamped verbally.

**Review by rickhanlonii — COMMENTED** [review]
✅ oops forgot to submit

File: packages/shared/ReactElementType.js (hunk 1 above)
Comment (gaearon) [reply packages/shared/ReactElementType.js:27]: https://github.com/facebook/react/pull/7001/files#r67037956 :)

**Comment by zthxxx** [comment]
@sebmarkbage  After released `react@19`, I found this pr breaking a series of packages like:

- https://github.com/zthxxx/react-dev-inspector
- https://github.com/infi-pc/locatorjs
- https://github.com/zh-lx/code-inspector
- https://github.com/ericclemmons/click-to-component

which depend on the `_debugSource` on fiber

**Comment by zthxxx** [comment]
`Error.stack` cannot retrieve the accurate absolute path locations of the source files.

They can only resolve to paths like webpack's virtual module paths (e.g., `webpack://`), 
and the line numbers correspond to the compiled/compressed code, not the original source files,
resulting in line number offsets that do not match the source code.

**Pull request merged**